src/semeio/forward_models/design_kw/design_kw.py

```diff
@@ -178,3 +184,41 @@ def rm_genkw_prefix(paramsdict, ignoreprefixes="LOG10_"):
         for keyvalue in keyvalues
         if keys.count(keyvalue[0]) == 1
     }
+
+
+def validate_configuration(
+    template_file_name: str, parameters_file_name: str = "parameters.txt"
+) -> None:
+    try:
+        if not os.path.exists(template_file_name):
+            raise ValueError(
```

Comment:
By raising and aborting at this point you only get the first warning that is encountered which can be frustrating for users. If you structure the validation like this:

```python
def validate_configuration(
    template_file_name: str, parameters_file_name: str = "parameters.txt"
) -> None:
    validate_template_file(template_file_name)
    validate_parameters_file(parameters_file_name)
```
you can create warnings both for the template file and the parameters file, when there are warnings for both.

Reply:
Agree with this one :+1: